mysql51/src/main/java/mysql51/badclassname.java

```diff
@@ -0,0 +1,24 @@
+package mysql51;
+
+public class badclassname {
+	public void ha() {
+		String str1 = "blue";
+		String str2 = "blue";
+		String str3 = str1;
+
+		if (str1 == str2)
+		{
+		  System.out.println("they're both 'blue'"); // this doesn't print because the objects are different
+		}
+
+		if (str1 == "blue")
+		{
+		  System.out.println("they're both 'blue'"); // this doesn't print because the objects are different
+		}
+
+		if (str1 == str3)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2583)